File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
# class Solution:
#   def strStr(self, haystack: str, needle: str) -> int:
#     return haystack.find(needle)

class Solution:
  def strStr(self, haystack: str, needle: str) -> int:
    p = 0
    l = len(needle)
    for p in range(len(haystack) - l + 1):
      if haystack[p:p + l] == needle:
        return p
    return -1

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
  a := []rune(haystack)
  b := []rune(needle)
  for i := range len(a) - len(b) + 1 {
    j := len(b) - 1
    for j >= 0 && a[i+j] == b[j] {
      j--
    }
    if j < 0 {
      return i
    }
  }
  return -1
}